server/src/main/java/org/opensearch/cluster/coordination/UnsafeBootstrapMasterCommand.java

```diff
@@ -122,7 +134,9 @@ protected void processNodePaths(Terminal terminal, Path[] dataPaths, int nodeLoc
         Settings persistentSettings = Settings.builder()
             .put(metadata.persistentSettings())
             .put(UNSAFE_BOOTSTRAP.getKey(), true)
+            .put(Metadata.SETTING_READ_ONLY_SETTING.getKey(), applyClusterReadOnlyBlock)
```

Comment:
if there is a cluster block already present in the cluster and user doesnt pass set it explicitly, then it will get removed unintentionally. It should add this block only if it is true.

Reply:
Ohh nice catch, let me fix it.